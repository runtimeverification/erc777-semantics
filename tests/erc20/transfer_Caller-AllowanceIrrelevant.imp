test.preamble();

// Testing transfer(To,Value) when To = caller, with allowance assigned
// specifically lower than Value.  So the allowance is irrelevant for transfer.
// The allowance is only relevant for transferFrom.

test.approve(caller, 20);   // set caller allowance for itself below 23
test.transfer(caller, 23);  // transfer 23 to itself, larger than allowance 20
