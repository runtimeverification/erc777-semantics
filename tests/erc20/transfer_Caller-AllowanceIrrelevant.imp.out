7 is approving allowance 20 for 7
7 is transferring 23 to 7
Balance of 7 is 70
Balance of 7 is 70
Allowance of 7 for 7 is 20
